articles/connections/nativesocial/facebook.md

```diff
@@ -0,0 +1,115 @@
+* **Step 3**: Use the Facebook SDK to retrieve the users's profile
+* **Step 4**: The application can then use the Facebook Session Info token to authenticate with Auth0.
+
+## Prerequisites
+
+Before you configure Native Facebook login for your native app via Auth0, you must:
+
+1. [Set up your application](/connections/social/facebook) with Facebook and as an Auth0 connection
+1. [Use the relevant Facebook SDK in your application](https://developers.facebook.com/docs/apis-and-sdks/)
+1. Create an application with Auth0 (if you have not already) and enable the Sign in with Facebook option in the [Dashboard > (Your Application) > Settings > Advanced Settings > Device Settings](${manage_url}).
+  ![Native Social Login Settings](/media/articles/connections/nativesocial/native-social-login.png)
+1. Follow the implementation details below
+
+## Implementation details
+
+As above, the process to authenticate a user profile using Native Facebook login is a four-step one, from your application's perspective:
+
+### Step 1
+
+The application authenticates a user via the Facebook SDK. It will obtain an Access Token from Facebook.
+
+### Step 2
+
+The application uses the Access Token to request a [Facebook Session Info Access Token](https://developers.facebook.com/docs/facebook-login/access-tokens/session-info-access-token).
+
+This request will look similar to the following:
+
+```
+GET https://graph.facebook.com/v5.0/oauth/access_token?
+grant_type=fb_attenuate_token&
+client_id=457704041391802&
+fb_exchange_token=<facebook_access_token>
+```
+
+and the response:
+
+```json
+{
+    "access_token": "XAAGgR4b...1lHWNCpqrAhcpoAZDZD",
+    "token_type": "bearer",
+    "expires_in": 5183924
+}
+```
+
+### Step 3
+
+The application needs to retrieve the user profile from Facebook using the Facebook SDK, which will end in a request similar to the following:
+
+```
+GET https://graph.facebook.com/v5.0/<facebook user id>?access_token=<facebook access token>&fields=email,name 
+```
+
+### Step 4
+
+The application can then use the session info Access Token and the Facebook user profile to authenticate with Auth0 by calling Auth0's `/oauth/token` endpoint using the Token Exchange flow with the `facebook-session-access-token` token type. If all goes well, Auth0 will return a normal response from the exchange, with the addition of the user profile.
+
+```
+POST https://${account.namespace}/oauth/token
+
+grant_type: 'urn:ietf:params:oauth:grant-type:token-exchange'
+subject_token_type: 'http://auth0.com/oauth/token-type/facebook-session-access-token'
+audience: 'your-api'
+scope: 'read:appointments openid profile email email_verified'
+subject_token: 'XAAGgR4b...1lHWNCpqrUHZAEtUuZAhcpoAZDZD'
+client_id: '${account.clientID}'
+user_profile: '{"email":"john@example.com", "Name":"John Doe"}'
```

Comment:
Would be great to point out that the this should be a json object, encoded as a string.